### ISSUE TITLE
[Bug]: PubSubIO writes from bounded PCollection cannot be more than 100 bytes

### ISSUE THREAD

**Issue by jonathan-lemos** [apache/beam#27000]
### What happened?

Writing to Pub/Sub currently results in the following issue:
```
javax.naming.SizeLimitExceededException: Pubsub message of length 220 exceeds maximum of 100 bytes, when considering the payload and attributes. See https://cloud.google.com/pubsub/quotas#resource_limits
```

This happens because [this call](https://github.com/apache/beam/blob/c0b23a7ea0bd7faf0bdecb2d16e8aed8d8af24a3/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsub/PubsubIO.java#L1413) passes in the maximum number of *messages* allowed in a batch, when it should pass in the maximum number of *bytes* allowed in a batch.

This issue only seems to present when the PCollection is bounded.

### Issue Priority

Priority: 2 (default / most bugs should be filed as P2)

### Issue Components

- [ ] Component: Python SDK
- [X] Component: Java SDK
- [ ] Component: Go SDK
- [ ] Component: Typescript SDK
- [X] Component: IO connector
- [ ] Component: Beam examples
- [ ] Component: Beam playground
- [ ] Component: Beam katas
- [ ] Component: Website
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner

**Comment by jonathan-lemos** [comment]
.take-issue

**Comment by bvolpato** [comment]
The issue was fixed and will be available in Beam 2.49.0.

The impact is specific to batch jobs that use PubsubIO, as the wrong validation is part of `PubsubBoundedWriter`.

Thanks for the fix @jonathan-lemos.
